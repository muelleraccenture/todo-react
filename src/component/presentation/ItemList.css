.item-list {
    flex: 1;
    background-color: #282c34;
    color: white;
    list-style-type: none;
}

.list-item {
    margin-top: 10px;
}
