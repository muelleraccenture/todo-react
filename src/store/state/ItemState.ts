export interface ItemState {
    readonly label: string,
    readonly complete: boolean,
}
