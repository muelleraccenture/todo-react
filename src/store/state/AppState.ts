import {ItemState} from "./ItemState";

export interface AppState {
    readonly items: ReadonlyArray<ItemState>
}
