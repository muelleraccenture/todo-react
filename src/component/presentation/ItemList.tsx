import React from 'react';
import {ItemState} from "../../store/state/ItemState";

export interface ItemListProps {
    readonly items: ReadonlyArray<ItemState>
}

export interface ItemListDispatchProps {

}

export const ItemList = ({items}: ItemListProps & ItemListDispatchProps) => {

    const listItems = items.map(item =>
        <li style={{
                marginTop: '10px',
                backgroundColor: '#B4FCF7',
                color: 'pink',
                listStyleType: "none",
        }}
            key={item.label}>
            <p>
                {item.label}
            </p>
            <p>
                {item.complete}
            </p>
        </li>
    )

    return <div>
                <ul>{listItems}</ul>
            </div>
}
