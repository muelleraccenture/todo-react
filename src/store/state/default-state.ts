import {AppState} from "./AppState";

export const DEFAULT_STATE: AppState = {
    items: [
        {label: "join experimental noise band", complete: false},
        {label: "become bilingual", complete: true},
    ]
}
