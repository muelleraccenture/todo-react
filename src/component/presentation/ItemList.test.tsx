import React from "react";
import {ItemList} from "./ItemList";
import {ItemState} from "../../store/state/ItemState";
import renderer, {ReactTestRenderer} from "react-test-renderer"


describe(ItemList.name, () => {
    let tree: ReactTestRenderer, items: ReadonlyArray<ItemState>

    beforeEach(() => {
        items = [
            {label: 'wake me up before I go go', complete: true},
            {label: 'never give you up ', complete: false},
        ]

        tree = renderer.create(<ItemList items={items}/>)
    })

    it('displays items', function () {
        expect(tree).toMatchSnapshot()
    });
})
