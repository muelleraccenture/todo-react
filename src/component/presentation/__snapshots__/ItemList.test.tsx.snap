// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemList displays items 1`] = `
<div>
  <ul>
    <li
      style={
        Object {
          "backgroundColor": "#B4FCF7",
          "color": "pink",
          "listStyleType": "none",
          "marginTop": "10px",
        }
      }
    >
      <p>
        wake me up before I go go
      </p>
      <p />
    </li>
    <li
      style={
        Object {
          "backgroundColor": "#B4FCF7",
          "color": "pink",
          "listStyleType": "none",
          "marginTop": "10px",
        }
      }
    >
      <p>
        never give you up 
      </p>
      <p />
    </li>
  </ul>
</div>
`;
